### PR TITLE
Clang's ASTImport now returns Expected<> Types

### PR DESCRIPTION
**Pull request by gmittert**
Clang b211b92b88 changed Import to return an Expected<QualType> so
updated the usage to match

This should fix the upstream-with-swift build.
https://github.com/llvm-mirror/clang/commit/b211b92b8811a6f926a11912ca97291d56486d2a

Clang was changed here:
git-svn-id: https://llvm.org/svn/llvm-project/cfe/trunk@360760
91177308-0d34-0410-b5e6-96231b3b80d8

### CODE REVIEW

File: source/Symbol/ClangASTContext.cpp
Comment (compnerd) [inline source/Symbol/ClangASTContext.cpp]: I would fold this into one:

`if (auto T = importer.import(ClangUtil::GetQualType(src))`

**Review by compnerd — COMMENTED** [review]
Please add a reference to the SVN revision in the commit message.

**Comment by gmittert** [comment]
Don't merge this quite yet, the swift-clang auto rebaser hasn't pulled the corresponding patch into swift-clang yet.

Edit: it's now in swift-clang upstream with swift.

**Comment by gmittert** [comment]
@swift-ci please test

**Comment by gmittert** [comment]
Do the tests actually work on upstream-with-swift PRs? The seem to keep failing on the same thing: something with swig on Linux, and something in swift on macos.

**Comment by gmittert** [comment]
CC: @dcci @adrian-prantl @xiaobai @JDevlieghere

**Comment by dcci** [comment]
@swift-ci test

File: source/Symbol/ClangASTContext.cpp
Comment (JDevlieghere) [inline source/Symbol/ClangASTContext.cpp:1446]: I believe this will assert in the destructor of llvm::Error when the expected is empty. You'd have to do something like   in CopyDecl: 

```
if (auto expected = importer.Import(ClangUtil::GetQualType(src))) {
  ...
} else { 
  Log *log = lldb_private::GetLogIfAllCategoriesSet(LIBLLDB_LOG_EXPRESSIONS);
  LLDB_LOG_ERROR(log, expected.takeError(), "Couldn't import type: {0}");	
}

```

**Review by JDevlieghere — CHANGES_REQUESTED** [review]

**Comment by dcci** [comment]
> Do the tests actually work on upstream-with-swift PRs? The seem to keep failing on the same thing: something with swig on Linux, and something in swift on macos.

`upstream-with-swift` is a best-effort branch. Things aren't guaranteed to always work there.
This is true for `master-next` in swift too.

**Comment by gmittert** [comment]
> > Do the tests actually work on upstream-with-swift PRs? The seem to keep failing on the same thing: something with swig on Linux, and something in swift on macos.
> 
> `upstream-with-swift` is a best-effort branch. Things aren't guaranteed to always work there.
> This is true for `master-next` in swift too.

Does that then mean it's fine to merge even if builds are failing?

File: source/Symbol/ClangASTContext.cpp
Comment (gmittert) [reply source/Symbol/ClangASTContext.cpp:1446]: I can add the logging, but calling operator bool() on it should mark it as checked if it's empty

https://llvm.org/doxygen/Support_2Error_8h_source.html#l00235

**Comment by gmittert** [comment]
Clang issue should now be fixed,

@swift-ci please test

**Comment by gmittert** [comment]
@swift-ci please test

File: source/Symbol/ClangASTContext.cpp
Comment (JDevlieghere) [reply source/Symbol/ClangASTContext.cpp:1446]: Yup, you are correct. I think the logging here is good, but otherwise it's customary to have an [`llvm::consumeError`](https://llvm.org/doxygen/namespacellvm.html#aee0e70be66536b9ef7b65ae2e5a45959) when you want to explicitly discarding the error.

**Pull request merged**
